chain/indexer.go

```diff
@@ -434,7 +435,15 @@ func (t *TipSetIndexer) TipSet(ctx context.Context, ts *types.TipSet) error {
 			res.Report[idx].StartedAt = res.StartedAt
 			res.Report[idx].CompletedAt = res.CompletedAt
 
-			if res.Report[idx].ErrorsDetected != nil {
+			if err := res.Report[idx].ErrorsDetected; err != nil {
+				// because error is just an interface it may hold a value of any concrete type that implements it, and if
+				// said type has unexported fields json marshaling will fail when persisting.
+				e, ok := err.(error)
+				if ok {
+					res.Report[idx].ErrorsDetected = &struct {
+						Error string
+					}{Error: e.Error()}
+				}
```

Comment:
Is it worth putting a placeholder for Error if it can't be extracted? Something like "invalid/empty error detail" or similar?

Reply:
Humm we could probably do that, but it would mean enforcing everything that writes to the struct filed being an error since it can contain types that do not implement the error interface:
https://github.com/filecoin-project/lily/blob/fe9a838028cb447ae54e8af0b5349d57c7912427/tasks/actorstate/task.go#L164-L170

I'm not sure what the best way to enforce that would be, I'm not sure if the `ErrorsDetected` filed can be type `error` due to go-pg. What do you think?

Reply:
I'm going to get this merged, but feel free to continue here if anything comes to mind.